bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/LegalEntityService.kt

```diff
@@ -70,16 +75,16 @@ class LegalEntityService(
         }
     }
 
-    fun getLegalEntities(limit: Int, startAfter: String?, externalIds: Collection<String>? = null): PageStartAfterResponse<LegalEntityGateInputResponse> {
-        val partnerCollection = saasClient.getLegalEntities(limit, startAfter, externalIds)
+    fun getLegalEntities(page: Int, size: Int, externalIds: Collection<String>? = null): PageLegalEntityResponse<LegalEntityGateInputResponse> {
 
-        val validEntries = toValidLegalEntities(partnerCollection.values)
-
-        return PageStartAfterResponse(
-            total = partnerCollection.total,
-            nextStartAfter = partnerCollection.nextStartAfter,
-            content = validEntries,
-            invalidEntries = partnerCollection.values.size - validEntries.size
+        val legalEntitiesPage = legalEntityRepository.findByExternalIdIn(externalIds, PageRequest.of(page, size))
```

Comment:
When I execute an endpoint request I get an error here, since the repo method does not expect null as a valid value. Should be empty instead